### PR TITLE
Add satisfy and enforce position limits utility functions

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This adds two utilities related to limits for checking if limits are within limits given an epsilon and another which force the provided position within the limits.

### CODE REVIEW

File: tesseract_common/include/tesseract_common/kinematic_limits.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+bool satisfiesPositionLimits(const Eigen::Ref<const Eigen::VectorXd>& joint_positions,
+                             const Eigen::Ref<const Eigen::MatrixX2d>& position_limits,
+                             double epsilon = std::numeric_limits<float>::epsilon());
+
+/**
+ * @brief Enforce position to be within the provided limits
+ * @param joint_positions The joint position to enforce bounds on
+ * @param joint_limits The limits to perform check
+ */
+void enforcePositionLimits(Eigen::Ref<Eigen::VectorXd> joint_positions,
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_common/include/tesseract_common/kinematic_limits.h:77]: This is currently how I am using internally. So the `satisfiesLimits` will check if the position is within the limits provided a given epsilon. Then the `enforceLimits` cleans up the data if it is outside the limits but within the epsilon. 

``` c++
if (satisfiesLimits(pos, limits))
  enforceLimits(pos, limits);
else()
  throw exception...
endif()
```

[the remark this answers is not in the document]
Comment (mpowelson) [reply tesseract_common/include/tesseract_common/kinematic_limits.h:77]: Ah. Ok I guess that is fine then

File: tesseract_common/src/kinematic_limits.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  return true;
+}
+
+void enforcePositionLimits(Eigen::Ref<Eigen::VectorXd> joint_positions,
+                           const Eigen::Ref<const Eigen::MatrixX2d>& position_limits)
+{
+  joint_positions = (joint_positions.array() > position_limits.col(1).array() ||
+                     joint_positions.array() < position_limits.col(0).array())
+                        .select(position_limits.col(1).array(), joint_positions.array());
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_common/src/kinematic_limits.cpp:58]: The limits when create will already be rounded to the machine epsilon so copying it should have the same values I believe. Your suggesting pushing the slightly inside the limits by epsilon?

File: tesseract_common/include/tesseract_common/kinematic_limits.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+namespace tesseract_common
+{
+/** @brief Store kinematic limits */
+struct KinematicLimits
+{
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  Eigen::MatrixX2d joint_limits;
+  Eigen::VectorXd velocity_limits;
+  Eigen::VectorXd acceleration_limits;
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_common/include/tesseract_common/kinematic_limits.h]: added.

File: tesseract_common/src/kinematic_limits.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_common/src/kinematic_limits.cpp:58]: @mpowelson I will merge but lets continue to discuss and I will create another PR if we make a change.

**Pull request merged**